### PR TITLE
initial commit of an ARM template solution to deploy App service web app with regional vnet integration and private endpoint connections to an Azure SQL DB and Azure Storage account

### PR DESCRIPTION
**Pull request by cbellee**
# Best Practice Checklist

Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment.
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/metadata.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "itemDisplayName": "301-web-app-regional-vnet-private-endpoint-sql-storage",
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/metadata.json:3]: use a "friendly" display name

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F301-web-app-regional-vnet-private-endpoint-sql-storage%2Fazuredeploy.json)    
+
+
+### Overview
+
+This solution deploys a Windows Azure App Services web app into a subnet using regional vnet integration. An Azure SQL DB and a storage account are also created, each with its own private link and private DNS resources to allow access form the web app over a private address space. FIrewall rules are added to the storage account ot prevent publci access & the Azure SQL Server is configured to disallow public traffic. A dns record is created in each zone for the Azure SQL Server and storage account.
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/README.md:19]: fix typos

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+        },
+        "webAppName": {
+            "type": "string",
+            "metadata": {
+                "description": "unique web app name"
+            }
+        },
+        "sqlAdministratorLoginName": {
+            "type": "string",
+            "defaultValue": "dbadmin",
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:34]: remove defaultValue, increases surface area

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "deployment location"
+            }
+        },
+        "webAppName": {
+            "type": "string",
+            "metadata": {
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:28]: add a globally unique defaultValue

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+        "privateLinkIpConfigsNestedTemplateUri": "[uri(parameters('_artifactsLocation'), concat('nestedtemplates/private_link_ipconfigs.json', parameters('_artifactsLocationSasToken')))]",
+        "privateLinkIpConfigsHelperNestedTemplateUri": "[uri(parameters('_artifactsLocation'), concat('nestedtemplates/private_link_ipconfigs_helper.json', parameters('_artifactsLocationSasToken')))]"
+
+    },
+    "resources": [
+        {
+            "comments": "This deployment links to the vnets.json nested ARM template",
+            "name": "[concat('linkedTemplate-vnet', '-', copyIndex())]",
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2018-05-01",
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:79]: update apiVersion

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+            "resourceGroup": "[resourceGroup().name]",
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:80]: remove this property

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+                        "value": "[variables('suffix')]"
+                    },
+                    "location": {
+                        "value": "[parameters('location')]"
+                    },
+                    "vNets": {
+                        "value": "[parameters('vNets')[copyIndex()]]"
+                    }
+                },
+                "templateLink": {
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:98]: move this above properties

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,458 @@
...
+                    "contentVersion": "1.0.0.0",
+                    "uri": "[variables('vnetNestedTemplateUri')]"
+                }
+            }
+        },
+        {
+            "comments": "This deployment links to the peerings nested ARM template",
+            "name": "linkedTemplate-peerings",
+            "type": "Microsoft.Resources/deployments",
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:107]: make the changes requested above on all /deployments resources throughout the template

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.parameters.json [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+                                        "serviceName": "Microsoft.Web/serverFarms"
+                                    }
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        },
+        "webAppName": {
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.parameters.json:50]: remove this param

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.parameters.json [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+            "value": "test-app"
+        },
+        "sqlAdministratorLoginName": {
+            "value": "dbadmin"
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.parameters.json:54]: GEN-UNIQUE

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/private_link_ipconfigs_helper.json [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    }
+  },
+  "variables": {
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2019-10-01",
+      "name": "[concat('nestedTemplate-private-link-ipconfigs-helper', copyIndex())]",
+      "properties": {
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/private_link_ipconfigs_helper.json:31]: move properties to the bottom of the resource definition

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/private_link_ipconfigs.json [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+      }
+    },
+    "variables": {
+    },
+    "resources": [
+      {
+        "type": "Microsoft.Resources/deployments",
+        "apiVersion": "2019-10-01",
+        "name": "private-link-ipconfigs",
+        "properties": {
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/private_link_ipconfigs.json:37]: move properties to the bottom of the resource definition

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/sqldb.json [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+                    "name": "[parameters('databaseName')]",
+                    "type": "databases",
+                    "apiVersion": "2019-06-01-preview",
+                    "location": "[parameters('location')]",
+                    "properties": {
+                        "edition": "[parameters('databaseEdition')]",
+                        "collation": "[parameters('databaseCollation')]",
+                        "requestedServiceObjectiveName": "[parameters('databaseServiceObjectiveName')]"
+                    },
+                    "dependsOn": [
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/sqldb.json:83]: move this above properties

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/sqldb.json [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+                {
+                    "name": "AllowAllMicrosoftAzureIps",
+                    "type": "firewallrules",
+                    "apiVersion": "2019-06-01-preview",
+                    "location": "[parameters('location')]",
+                    "properties": {
+                        "endIpAddress": "0.0.0.0",
+                        "startIpAddress": "0.0.0.0"
+                    },
+                    "dependsOn": [
```
Comment (bmoore-msft) [inline 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/sqldb.json:96]: move this above properties

Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/sqldb.json:96]: Ok, I have made the requested change.

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/sqldb.json (hunk 13 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/sqldb.json:83]: Ok, I have made the requested change.

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/private_link_ipconfigs.json (hunk 12 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/private_link_ipconfigs.json:37]: Ok, I have made the requested change.

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/private_link_ipconfigs_helper.json (hunk 11 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/nestedtemplates/private_link_ipconfigs_helper.json:31]: Ok, I have made the requested change.

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.parameters.json (hunk 10 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.parameters.json:54]: Ok, I have made the requested change.

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json (hunk 8 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:107]: Ok, I have made the requested changes.

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json (hunk 7 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:98]: Ok, I have made the requested change.

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json (hunk 5 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:79]: Ok, I have made the requested changes.

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json (hunk 4 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:28]: Added a the default value below. 

"defaultValue": "[concat('web-app-', uniqueString(subscription().id, resourceGroup().id))]",

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json (hunk 3 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:34]: removed "defaultValue"

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/README.md (hunk 2 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/README.md:19]: Fixed typos

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/metadata.json (hunk 1 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/metadata.json:3]: added friendly display name

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json (hunk 6 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.json:80]: Hi Brian, not sure what I need to do here. Which property are you referring to?

File: 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.parameters.json (hunk 9 above)
Comment (cbellee) [reply 301-web-app-regional-vnet-private-endpoint-sql-storage/azuredeploy.parameters.json:50]: Parameter removed.

**Pull request merged**